### PR TITLE
fix: bail out of `firebase init hosting:github` early with helpful message if Hosting isn't set up

### PR DESCRIPTION
**Pull request by devpeerapong**
### Description

#3113 

### Scenarios Tested

1. Create an empty project
2. Run `firebase init hosting:github`
3. The CLI should bail out before the authentication process with Firebase and Github

### CODE REVIEW

File: src/init/features/hosting/github.ts [hunk 1]
```diff
@@ -54,6 +54,13 @@ export async function initGitHub(setup: Setup, config: any, options: any): Promi
     return reject("Could not determine Project ID, can't set up GitHub workflow.", { exit: 1 });
   }
 
+  if (!setup.config.hosting) {
+    return reject(
+      `The project is not setup with Hosting, Please use ${bold("firebase init hosting")} instead`,
```
Comment (jhuleatt) [inline src/init/features/hosting/github.ts:59]:
```suggestion
      `Didn't find a Hosting config in firebase.json. Run ${bold("firebase init hosting")} instead.`,
```

**Comment by jhuleatt** [comment]
Thank you @devpeerapong for your help! I suggested a change to the language in the error message. Can you also please add a line in [`CHANGELOG.md`](https://github.com/firebase/firebase-tools/blob/master/CHANGELOG.md?plain=1) as part of this PR?

**Comment by devpeerapong** [comment]
both error message and `CHANGELOG.md` are updated.

**Review by jhuleatt — APPROVED** [review]

File: src/init/features/hosting/github.ts [hunk 2]
```diff
@@ -54,6 +54,13 @@ export async function initGitHub(setup: Setup, config: any, options: any): Promi
     return reject("Could not determine Project ID, can't set up GitHub workflow.", { exit: 1 });
   }
 
+  if (!setup.config.hosting) {
+    return reject(
+      `Didn't find a Hosting config in firebase.json. Run ${bold("firebase init hosting")} instead.`,
```
Comment (bkendall) [inline src/init/features/hosting/github.ts:59]: Please run `npm run format` in the codebase to clear up this error

File: src/init/features/hosting/github.ts [hunk 3]
```diff
@@ -54,6 +54,13 @@ export async function initGitHub(setup: Setup, config: any, options: any): Promi
...
+      { exit: 1 }
```
Comment (bkendall) [inline src/init/features/hosting/github.ts:60]: `exit: 1` is the default and can be omitted.

**Review by bkendall — CHANGES_REQUESTED** [review]

**Review by bkendall — APPROVED** [review]

**Pull request merged**